### PR TITLE
storage: added host-flash module

### PR DESCRIPTION
**Pull request by Diabliz**

### CODE REVIEW

**Comment by nalajcie** [comment]
Bam, another case of #76

File: storage/host-flash/host-flash.c [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+    while (wrote != bufflen) {
+        stat = hostflash_read(addr + wrote, &currentByte, 1);
+        if (stat <= 0)
+            return stat;
+
+        currentByte = currentByte & *((char *)buff + wrote);
+        stat = pwrite(hostflash_common.filefd, &currentByte, 1, addr + wrote);
+        if (stat < 0) {
+            if(errno == -EINTR)
```
Comment (agkaminski) [inline storage/host-flash/host-flash.c:69]: missing space between `if` and `(`

File: storage/host-flash/host-flash.c [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    int erased = 0;
+    unsigned int sectorNum;
+    unsigned int sectorAddr;
+    int stat;
+
+    if (addr < hostflash_common.flashsz) {
+        sectorNum = addr / hostflash_common.sectorsz;
+        sectorAddr = sectorNum * hostflash_common.sectorsz;
+    }
+    else
```
Comment (agkaminski) [inline storage/host-flash/host-flash.c:94]: if `if` have brackets, `else` should have them too.

File: storage/host-flash/host-flash.c [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        return;
+
+    while (erased != hostflash_common.sectorsz) {
+        stat = pwrite(hostflash_common.filefd, &tempVar, 1, sectorAddr + erased);
+        if (stat < 0) {
+            if(errno == -EINTR)
```
Comment (agkaminski) [inline storage/host-flash/host-flash.c:100]: missing space between `if` and `(`

File: storage/host-flash/host-flash.c [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+void hostflash_chipErase(void)
+{
+    char tempTab[16] = {0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+                        0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF};
+    size_t erased = 0, len = sizeof(tempTab);
+    ssize_t stat;
+
+    while (erased != hostflash_common.flashsz) {
+        stat = pwrite(hostflash_common.filefd, tempTab, len, erased);
```
Comment (agkaminski) [inline storage/host-flash/host-flash.c:121]: Why not make tempTab bigger and go with (for example):
```
char tempTab[256];
memset(tempTab, 0xff, sizeof(tempTap);
```

Bigger chunks, better performance.

File: storage/host-flash/host-flash.c [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        if (stat < 0) {
+            if(errno == -EINTR)
+                continue;
+
+            return;
+        } else {
```
Comment (agkaminski) [inline storage/host-flash/host-flash.c:127]: `else {` should be in the new line

File: storage/host-flash/host-flash.c [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+{
+    char tempTab[16] = {0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+                        0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF};
+    size_t erased = 0, len = sizeof(tempTab);
+    ssize_t stat;
+
+    while (erased != hostflash_common.flashsz) {
+        stat = pwrite(hostflash_common.filefd, tempTab, len, erased);
+        if (stat < 0) {
+            if(errno == -EINTR)
```
Comment (agkaminski) [inline storage/host-flash/host-flash.c:123]: missing space between `if` and `(`

File: storage/host-flash/host-flash.c [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+
+    if (addr < hostflash_common.flashsz) {
+        sectorNum = addr / hostflash_common.sectorsz;
+        sectorAddr = sectorNum * hostflash_common.sectorsz;
+    }
+    else
+        return;
+
+    while (erased != hostflash_common.sectorsz) {
+        stat = pwrite(hostflash_common.filefd, &tempVar, 1, sectorAddr + erased);
```
Comment (agkaminski) [inline storage/host-flash/host-flash.c:98]: Not really best way to write data, byte by byte. Analogous way that chip erase is performed would be better.

File: storage/host-flash/host-flash.c [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+        return -EINVAL;
+
+    hostflash_common.flashsz = *flashsz;
+    hostflash_common.sectorsz = *sectorsz;
+    hostflash_common.filefd = open(fileName, O_CREAT | O_RDWR, 0666);
+    if (hostflash_common.filefd < 0)
+        return hostflash_common.filefd;
+    stat = ftruncate(hostflash_common.filefd, hostflash_common.flashsz);
+    if (stat < 0)
+        return stat;
```
Comment (agkaminski) [inline storage/host-flash/host-flash.c:154]: We might want to close the file here

File: storage/host-flash/host-flash.c [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+
+    return;
+}
+
+
+int hostflash_init(size_t *flashsz, size_t *sectorsz, const char *fileName)
+{
+    int stat;
+
+    if ((*sectorsz % 2) || (*flashsz % *sectorsz))
```
Comment (agkaminski) [inline storage/host-flash/host-flash.c:144]: `sectorsz` bigger than `flashsz` would pass here ;) Same goes for `flashsz` and `sectorsz` == 0

File: storage/host-flash/host-flash.c (hunk 1 above)
Comment (Diabliz) [reply storage/host-flash/host-flash.c]: done

**Comment by agkaminski** [comment]
`hostflash_write` needs rework - we can do much better than writing byte by byte.

**Pull request merged**